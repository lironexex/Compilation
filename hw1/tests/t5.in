// Integers - BIN_INT, OCT_INT:
var num1 = 0b1010001 + 0b0 * 0000b1010 + 0b
var num2 = 000o700 + 0o800 + 0o + 0o700
0b0000000000101
0o0000000000009
0B10
0O70
