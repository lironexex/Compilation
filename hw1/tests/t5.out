1 COMMENT 1
2 VAR var
2 ID num1
2 ASSIGN =
2 BIN_INT 81
2 BINOP +
2 BIN_INT 0
2 BINOP *
2 DEC_INT 0
2 ID b1010
2 BINOP +
2 DEC_INT 0
2 ID b
3 VAR var
3 ID num2
3 ASSIGN =
3 DEC_INT 0
3 ID o700
3 BINOP +
3 DEC_INT 0
3 ID o800
3 BINOP +
3 DEC_INT 0
3 ID o
3 BINOP +
3 OCT_INT 448
4 BIN_INT 5
5 OCT_INT 0
5 DEC_INT 9
6 DEC_INT 0
6 ID B10
7 DEC_INT 0
7 ID O70
